src/FormBuilder/resources/js/form-builder/src/components/EmbedForm/index.tsx

```diff
@@ -0,0 +1,542 @@
+    createPostType: string;
+    currentPostType: string;
+    newPostName: string;
+    selectedPost: string;
+    selectedStyle: string;
+    openFormButton: string;
+    isCopied: boolean;
+    isInserting: boolean;
+    isCreating: boolean;
+    isInserted: boolean;
+    isCreated: boolean;
+    createdLink: string;
+    insertedLink: string;
+}
+
+/**
+ * @unreleased
+ */
+export default function EmbedFormModal({handleClose}: EmbedFormModalProps) {
+
+    const {formId} = getWindowData();
+    const newPostNameRef = useRef<HTMLInputElement>(null);
+    const openFormBtnRef = useRef<HTMLInputElement>(null);
+
+    const [state, setState] = useState<StateProps>({
+        posts: [],
+        insertPostType: 'page',
+        createPostType: 'page',
+        currentPostType: 'page',
+        newPostName: '',
+        selectedPost: '',
+        selectedStyle: 'onpage',
+        openFormButton: '',
+        isCopied: false,
+        isInserting: false,
+        isCreating: false,
+        isInserted: false,
+        isCreated: false,
+        createdLink: null,
+        insertedLink: null,
+    });
+
+    const {editEntityRecord, saveEditedEntityRecord, saveEntityRecord} = useDispatch(store);
+
+    const closeModal = useCallback(e => {
+        if (e.keyCode === 27) {
+            handleClose(e);
+        }
+    }, []);
+
+    useEffect(() => {
+        document.addEventListener('keydown', closeModal, false);
+        return () => document.removeEventListener('keydown', closeModal, false);
+    }, []);
+
+    const postOptions = [
+        {label: 'Page', value: 'page'},
+        {label: 'Post', value: 'post'},
+    ];
+
+    const displayStyles = [
+        {
+            label: __('Full form', 'give'),
+            value: 'onpage',
+            description: __('All fields are visible on one page with the donate button at the bottom', 'give'),
+        },
+        {
+            label: __('Modal', 'give'),
+            value: 'modal',
+            description: __('Add description for Modal', 'give'),
+        },
+        {
+            label: __('New Tab', 'give'),
+            value: 'new-tab',
+            description: __('Add description for New tab', 'give'),
+        },
+    ];
+
+    // Fetch posts/pages
+    const isLoadingPages = useSelect((select) => {
+        const filtered: Array<Post> = [];
+        // @ts-ignore
+        const data = select(store).getEntityRecords('postType', state.currentPostType, {
+            status: ['publish', 'draft'],
+            per_page: -1, // do we want this?
+        });
+
+        if (data) {
+            data?.forEach(page => {
+                filtered.push({
+                    value: page.id,
+                    label: page.title.raw || __('(no title)', 'give'),
+                    content: page.content.raw,
+                    disabled: page.content.raw.includes(`wp:give/donation-form {"id":${formId}`), // disable pages that already have this form block included
+                });
```

Comment:
can you explain why we need to "disable pages that already have this form block included"?

Reply:
I thought that having the two identical forms was something that nobody needs, but I may be wrong. We can easily remove that.

Reply:
Great thoughts! I appreciate the cleverness here, I would just be concerned people wouldn't understand this without an explanation so maybe better to remove for now.

Reply:
Removed